src/main/java/com/google/javascript/gents/ModuleConversionPass.java

```diff
@@ -324,14 +324,19 @@ void convertRequireToImportStatements(Node n, String fullLocalName, String requi
 
     String referencedFile = pathUtil.getImportPath(n.getSourceFileName(), module.file);
 
-    // Uses default import syntax as this is a javascript namespace
+    // This is a javascript namespace
     if (module.shouldUseOldSyntax()) {
+      Node nodeToImport = null;
+      // If it has a default export then use `import foo from "goog:bar";`
+      if (module.hasDefaultExport) {
+        nodeToImport = Node.newString(Token.NAME, localName);
+      } else {
+        // If it doesn't have a default export then use `import * as foo from "goog:bar";`
+        nodeToImport = Node.newString(Token.IMPORT_STAR, localName);
+      }
```

Comment:
what about destructuring imports. At least add a todo for me to look into that here.

Reply:
I tried to handle destructing imports but got lost and broke many unrelated tests.
I added a TODO and would like solve it in a different PR with a clean branch.